pymc_marketing/mmm/plot.py

```diff
@@ -235,15 +346,114 @@ def random_samples(
     ]
 
 
+def generate_colors(n: int, start: int = 0) -> list[str]:
+    """Generate list of colors."""
+    return [f"C{i}" for i in range(start, start + n)]
+
+
+def _general_plot_function(
+    curve: xr.DataArray,
+    non_grid_names: set[str],
+    get_plot_data: GetPlotData,
+    make_selection: MakeSelection,
+    plot_selection: PlotSelection,
+    subplot_kwargs: dict | None = None,
+    axes: npt.NDArray[Axes] | None = None,
+    same_axes: bool = False,
+    colors: Iterable[str] | None = None,
+    legend: bool = False,
+    plot_kwargs: dict[str, Any] | None = None,
+    patch: bool = True,
+    line: bool = True,
+    sel_to_string: SelToString | None = None,
+) -> tuple[plt.Figure, npt.NDArray[Axes]]:
```

Comment:
I think this auxiliary function would benefit from doc-strings

Reply:
Added a bit to it for some context